### PR TITLE
Adds Eigen pitfalls link

### PR DESCRIPTION
**Pull request by ysims**
Addresses #115.

Adds a link to the common Eigen pitfalls page in the links section.

Also mentions the link the first time Eigen is introduced.

[Preview: Mathematics.](https://deploy-preview-143--nubook.netlify.app/system/foundations/mathematics)
[Preview: Glossary.](https://deploy-preview-143--nubook.netlify.app/guides/general/glossary)

### CODE REVIEW

File: src/book/02-system/01-foundations/05-mathematics.mdx [hunk 1]
```diff
@@ -10,7 +10,7 @@ The NUbots codebase uses specific mathematical concepts and notation that are re
 
 ## Linear Algebra
 
-We currently use the [Eigen](https://eigen.tuxfamily.org/) C++ template library for linear algebra in the NUbots codebase.
+We currently use the [Eigen](https://eigen.tuxfamily.org/) C++ template library for linear algebra in the NUbots codebase. Be aware that Eigen has some unique pitfalls and these are listed on the [Eigen pitfalls page](https://eigen.tuxfamily.org/dox/TopicPitfalls.html).
```
Comment (ysims) [inline src/book/02-system/01-foundations/05-mathematics.mdx:13]:
```suggestion
We currently use the [Eigen](https://eigen.tuxfamily.org/) C++ template library for linear algebra in the NUbots codebase. Be aware that Eigen has some common pitfalls that are listed on the [Eigen pitfalls page](https://eigen.tuxfamily.org/dox/TopicPitfalls.html).
```

**Comment by JosephusPaye** [comment]
Does it make sense to link to common pitfalls in the Eigen glossary entry as well?

```diff
--Eigen: The C++ linear algebra library NUbots uses. Documentation can be found [here](url).
++Eigen: The C++ linear algebra library NUbots uses. Documentation can be found [here](url). List of common pitfalls can be found [here](url).
```

**Comment by KipHamiltons** [comment]
> Does it make sense to link to common pitfalls in the Eigen glossary entry as well?
> 
> ```diff
> --Eigen: The C++ linear algebra library NUbots uses. Documentation can be found [here](url).
> ++Eigen: The C++ linear algebra library NUbots uses. Documentation can be found [here](url). List of common pitfalls can be found [here](url).
> ```

Great idea! I think it's hard to overestimate the importance of these tbh

**Comment by ysims** [comment]
Added to glossary @KipHamiltons @JosephusPaye.

**Review by KipHamiltons — APPROVED** [review]

**Pull request merged**